### PR TITLE
Various issues from testing

### PR DESCRIPTION
**Pull request by mikolak**

### CODE REVIEW

File: example/lib/test_scenarios/sensor_tag_test_scenario.dart [hunk 1]
```diff
@@ -21,7 +21,8 @@ class SensorTagTestScenario {
         .then((_) => _peripheralTestOperations.readWriteMonitorCharacteristic())
         .then((_) => _peripheralTestOperations.fetchConnectedDevice())
         .then((_) => _peripheralTestOperations.fetchKnownDevice())
-        .then((_) => _peripheralTestOperations.disconnect());
+        .then((_) => _peripheralTestOperations.disconnect())
+        .catchError((error) => _peripheralTestOperations.logError(error));
```
Comment (pawelByszewski) [inline example/lib/test_scenarios/sensor_tag_test_scenario.dart]: it looks weird, is `logError` a proper method of peripheral test operations?

File: lib/blemulator/example/example_peripheral.dart [hunk 2]
```diff
@@ -17,17 +14,9 @@ class SensorTag extends SimulatedPeripheral {
             id: id,
             advertisementInterval: Duration(milliseconds: 800),
             services: [
-              SimulatedService(
+              TemperatureService(
```
Comment (pawelByszewski) [inline lib/blemulator/example/example_peripheral.dart]: it makes the sample more difficult to read for a developer that sees it the first time. maybe we should add more sophisticated device as second device?

File: lib/error/ble_error.dart [hunk 3]
```diff
@@ -49,3 +49,54 @@ class BleError {
       "characteristic UUID: $characteristicUUID, "
       "descriptor UUID: $descriptorUUID)";
 }
+
+class BleErrorCode {
```
Comment (pawelByszewski) [inline lib/error/ble_error.dart]: is it moved from somewhere or is it completely new piece of code?

File: lib/blemulator/example/example_peripheral.dart [hunk 4]
```diff
@@ -45,75 +34,73 @@ class SensorTag extends SimulatedPeripheral {
...
-          value: Uint8List.fromList([101, 254, 64, 12]),
-          convenienceName: "IR Temperature Data",
-          isNotifiable: true,
-        );
-}
+  void _emitTemperature() async {
+    while (true) {
+      Uint8List delayBytes =
+          await characteristicByUuid(_temperaturePeriodUuid).read();
+      int delay = delayBytes.buffer.asByteData().getUint8(0) * 10;
```
Comment (bartoszwilk) [inline lib/blemulator/example/example_peripheral.dart]: Shouldn't it be getUint8(8)?

File: lib/blemulator/example/example_peripheral.dart [hunk 5]
```diff
@@ -45,75 +34,73 @@ class SensorTag extends SimulatedPeripheral {
...
+                convenienceName: "IR Temperature Config",
+              ),
+              SimulatedCharacteristic(
+                  uuid: _temperaturePeriodUuid,
+                  value: Uint8List.fromList([80]),
+                  convenienceName: "IR Temperature Period"),
+            ],
+            convenienceName: convenienceName) {
+    characteristicByUuid(_temperatureConfigUuid).monitor().listen((value) {
+      int valueAsInt = value.buffer.asByteData().getUint8(8);
```
Comment (bartoszwilk) [inline lib/blemulator/example/example_peripheral.dart]: valueAsInt is not used

File: lib/blemulator/example/example_peripheral.dart [hunk 6]
```diff
@@ -45,75 +34,73 @@ class SensorTag extends SimulatedPeripheral {
...
+      _readingTemperature = value[0] == 1 ? true : false;
```
Comment (bartoszwilk) [inline lib/blemulator/example/example_peripheral.dart]: Why not just value[0] == 1?

**Pull request closed without merging**